### PR TITLE
Add option to wait for connection to `/v2/events` on gRPC/HTTP server

### PR DESCRIPTION
**Pull request by MarlonGamez**
**Description**
This PR adds the option for the user to have skaffold pause execution on initialization of the gRPC/HTTP server. Execution will continue when a request to `/v2/events` is made. This will help ensure that users of the server can connect before Skaffold execution ends early in some cases.

This PR also moves the call to `event.SendErrorMessageOnce` to the server shutdown callback function, which ensures that the error message gets sent out before the server is shut down.

### CODE REVIEW

File: cmd/skaffold/app/cmd/flags.go [hunk 1]
```diff
@@ -182,6 +182,15 @@ var flagRegistry = []Flag{
 		DefinedOn:     []string{"dev", "build", "run", "debug", "deploy", "render", "apply", "test"},
 		IsEnum:        true,
 	},
+	{
+		Name:          "wait-for-connection",
+		Usage:         "Blocks execution until the /v2/events gRPC/HTTP endpoint is hit",
+		Value:         &opts.WaitForConnection,
+		DefValue:      false,
+		FlagAddMethod: "BoolVar",
+		DefinedOn:     []string{"dev", "debug"},
```
Comment (aaron-prindle) [inline cmd/skaffold/app/cmd/flags.go:191]: Should the commands this is `DefinedOn` here be the same as those for `--enable-rpc`?  Might make sense as any command can initialize the gRPC/HTTP server and emit events

File: cmd/skaffold/skaffold.go [hunk 2]
```diff
@@ -57,7 +55,7 @@ func main() {
 			// for the entire skaffold run here. It's possible SetupColors() was never called, so call it again
 			// before we print an error to get the right coloring.
 			errOut := output.SetupColors(context.Background(), os.Stderr, output.DefaultColorCode, false)
-			eventV2.SendErrorMessageOnce(constants.DevLoop, constants.SubtaskIDNone, err)
+			//eventV2.SendErrorMessageOnce(constants.DevLoop, constants.SubtaskIDNone, err)
```
Comment (aaron-prindle) [inline cmd/skaffold/skaffold.go:58]: remove comment?

**Review by aaron-prindle — APPROVED** [review]
LGTM, seems this flag could be defined on more commands and it looks like a commented out line was left (see comments)

File: cmd/skaffold/app/cmd/flags.go (hunk 1 above)
Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/flags.go:191]: yeah, this was definitely a brain fart of mine 😅 thanks for catching

File: cmd/skaffold/skaffold.go (hunk 2 above)
Comment (MarlonGamez) [reply cmd/skaffold/skaffold.go]: thanks for catching this !

File: pkg/skaffold/event/v2/event.go [hunk 3]
```diff
@@ -173,6 +180,9 @@ func (ev *eventHandler) forEach(listeners *[]*listener, log *[]proto.Event, lock
 }
 
 func (ev *eventHandler) forEachEvent(callback func(*proto.Event) error) error {
+	if len(handler.wait) != cap(handler.wait) {
```
Comment (gsquared94) [inline pkg/skaffold/event/v2/event.go:183]: you can run into a race condition from multiple goroutines and have a deadlock here.

Comment (MarlonGamez) [reply pkg/skaffold/event/v2/event.go]: Updated this to use a select statement instead

**Review by nkubala — APPROVED** [review]

**Pull request merged**